app/models/job.rb

```diff
@@ -1,4 +1,46 @@
 class Job < ApplicationRecord
   validates :title, :level_of_interest, :city, presence: true
   belongs_to :company
+  belongs_to :category
+  has_many :comments
+
+  def sorted_comments
+    comments.order(:created_at).reverse
+  end
+
+  def self.sort_by(filter)
+    return order(:city) if filter == "location"
+    order(level_of_interest: :desc) if filter == "interest"
+  end
+
+  def self.by_location(city)
+    where(city: city)
+  end
+
+  def self.interest_counts
+    interest_level_data = []
+    interest_level_counts.each_pair do |k, v|
+      interest_level_data << {level: k, job_count: v}
+    end
```

Comment:
`each_pair` seems to add a lot of Ruby to this call. I would consider digging deeper on the ActiveRecord to determine if you can return a collection of objects that respond to `level` and `job_count`. Remember, in AR, if you can create a column with a name in your query that column name is then available to you as a method call on objects that are returned.

I know that you're not getting back objects, but that's because you're using `count` on the end of your DB call. Move that into a `select` and you should be able to move back to the world of objects.

I might want to benchmark the two options to see which is actually faster. I don't like the idea of using arrays of hashes in my view (prefer objects), but if there's a noticeable speed difference I could be convinced.